src/zenml/utils/pipeline_docker_image_builder.py

```diff
@@ -454,13 +474,68 @@ def _gather_requirements_files(
                     ", ".join(f"`{r}`" for r in integration_requirements_list),
                 )
 
+        if hub_requirements:
+            hub_requirements_file = "\n".join(hub_requirements)
+            requirements_files.append(
+                (
+                    ".zenml_hub_requirements",
+                    hub_requirements_file,
+                )
+            )
+            if log:
+                logger.info(
+                    "- Including hub requirements: %s",
+                    ", ".join(f"`{r}`" for r in hub_requirements),
+                )
+
         return requirements_files
 
+    @staticmethod
+    def _get_hub_requirements(
+        required_hub_plugins: List[Union[str, Tuple[str, str]]]
+    ) -> Tuple[Dict[str, List[str]], List[str]]:
+        """Gets the requirements for the hub plugins.
+
+        Args:
+            required_hub_plugins: List of hub plugin names.
+
+        Returns:
+            - Extra index requirements (the hub plugins themselves), mapping
+                index URLs to lists of package names
+            - List of dependencies of all required hub plugins
+        """
+        from zenml.cli.hub import _get_plugin  # TODO: move this to utils?
+
+        hub_packages: defaultdict[str, List[str]] = defaultdict(list)
+        hub_requirements: List[str] = []
+
+        for required_plugin in required_hub_plugins:
+            if isinstance(required_plugin, str):
+                plugin_name = required_plugin
+                plugin_version = "latest"
+            else:
+                plugin_name, plugin_version = required_plugin
+            plugin = _get_plugin(plugin_name, plugin_version)
+            if plugin and plugin.index_url and plugin.wheel_name:
+                hub_packages[plugin.index_url].append(plugin.wheel_name)
+                if plugin.requirements:
+                    hub_requirements.extend(plugin.requirements)
```

Comment:
What if the plugin requires another plugin?

Reply:
this is not possible at the moment in general, we do have an open ticket about supporting this but it's probably not high enough prio to make it into the MVP release